src/background/completed-test-step-telemetry-creator.ts

```diff
@@ -111,6 +117,16 @@ export class CompletedTestStepTelemetryCreator {
         return numInstances;
     }
 
+    private getStepDetails(
+        step: Requirement,
+        assessment: Assessment,
+        storeData: AssessmentStoreData,
+    ): any {
```

Comment:
The usage of the `any` type feels a bit off to me, since it's generally avoided in our code base. While I know the data returned is surely going to vary from step to step, it might be worth making a flexible type that includes common data like the result status (pass/fail/incomplete) for the step, similar to types we use for `telemetryData` and `payload` objects, and then make a specific `LabelInNameStepDetails` type that lists exactly what we'll be sending over for this test step.

Reply:
For payloads, we have a strict contract since we need to programmatically produce and consume these items. I'm also not convinced that we have a generic base that needs to always be in this type. In this case, it's just an object. I could use `Object`, but that's only marginally improved from `any`. A quick scan for `: any` reveals that we use it in a lot of places, so I'm not fully convinced that stronger typing is needed.